tests/shared/src/test/scala/scala/meta/tests/parsers/PositionSuite.scala

```diff
@@ -29,7 +29,7 @@ class PositionSuite extends ParseSuite {
     val tree = term("(1 + 2).foo")
     assert(tree.show[Positions] === """
       |Term.Select{0..11}(
-        |Term.ApplyInfix{0..7}(
+        |Term.ApplyInfix{1..6}(
```

Comment:
I'm not sure if this is good expected output since the parentheses are required here. The motivation to open #1083 was that redundant parentheses are included in the position for example in `(a).b`. In `(1 + 2).foo` the parens are required to construct that Term.ApplyInfix. I had imagined that the fix for this ticket would involve adapting `def termName` constructor for Term.Name nodes.

Any thoughts @xeno-by ?

Reply:
@olafurpg the parenthesis would not be necessary for the `Term.ApplyInfix` if we move it outside of the `Term.Select`.

Reply:
This PR handles infix applications, but what about other nodes?

```scala
@ val Term.Select(qual, _) = "(x match { case _ => 2 }).b".parse[Term].get
qqual: Term = Term.Match(Term.Name("x"), List(Case(_, None, Lit.Int(2))))

@ qual.tokens.syntax
res3: String = "(x match { case _ => 2 })"
```